homework/index.js

```diff
@@ -36,8 +36,92 @@
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const navDiv = createAndAppend('div', root, { class: 'nav' });
+        const navHeader = createAndAppend('div', navDiv, { class: 'nav-header' });
+        const navTitle = createAndAppend('div', navHeader, { class: 'nav-title' });
+        navTitle.innerText = 'HYF Repositories';
+        const selectEl = createAndAppend('select', navDiv, { id: 'getRepoData' });
+
+        data.sort((a, b) => (a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1));
+
+        data.forEach(res => {
+          createAndAppend('option', selectEl, { text: res.name, value: res.id });
+        });
       }
+      const selectRepo = document.getElementById('getRepoData');
+      const container = createAndAppend('div', root, { class: 'container' });
+
+      selectRepo.addEventListener('change', () => {
+        fetchJSON(url, (error, repData) => {
+          container.innerHTML = '';
+          const filteredData = repData.filter(el => el.id === Number(selectRepo.value));
+          const leftSection = createAndAppend('div', container, { class: 'repInfo-side' });
+          const repositoryName = createAndAppend('div', leftSection, { class: 'repName' });
+          const repositoryDescription = createAndAppend('div', leftSection, {
+            class: 'repDescription',
+          });
+
+          const repositoryForks = createAndAppend('div', leftSection, { class: 'repForks' });
+          const repositoryDate = createAndAppend('div', leftSection, { class: 'repDate' });
+
+          filteredData.forEach(allData => {
+            createAndAppend('span', repositoryName, { text: 'Repository Name:' });
```

Comment:
When you think of splitting your function into smaller functions, moving this block to a separate function would also promote re-usability. This way, for your first requirement where you need to display data for the first dropdown item by default and displaying updated data for different repos would be easier using the new function. Please let me know if you have any questions. 🙂